acme/api/middleware.go

```diff
@@ -324,6 +317,23 @@ func lookupJWK(next nextHTTP) nextHTTP {
 				render.Error(w, acme.NewError(acme.ErrorUnauthorizedType, "account is not active"))
 				return
 			}
+
+			if storedLocation := acc.GetLocation(); storedLocation == "" {
+				// Old ACME accounts may not have a stored location.
+				if !strings.HasPrefix(kid, kidPrefix) {
+					render.Error(w, acme.NewError(acme.ErrorMalformedType,
+						"kid does not have required prefix; expected %s, but got %s",
+						kidPrefix, kid))
+					return
+				}
+			} else if kid != storedLocation {
```

Comment:
Does it makes sense to use `!strings.EqualFold(kid, storeLocation)`?
If the kid is has base64 perhaps not.

Reply:
I think we should store the full kid instead of constructing it. We're just saving a few bytes and we're adding more complexity.